biz.aQute.tester.junit-platform/src/aQute/tester/junit/platform/Activator.java

```diff
@@ -0,0 +1,377 @@
+		thread = Thread.currentThread();
+
+		// Make sure that this is loaded
+		ClassLoader l = thread.getContextClassLoader();
+		try {
+			thread.setContextClassLoader(BundleEngine.class.getClassLoader());
+			launcher = LauncherFactory.create();
+		} catch (Throwable e) {
+			error("Couldn't load the BundleEngine: %s", e);
+			System.exit(254);
+		} finally {
+			thread.setContextClassLoader(l);
+		}
+
+		List<TestExecutionListener> listenerList = new ArrayList<>();
+
+		String testcases = context.getProperty(TESTER_NAMES);
+		trace("test cases %s", testcases);
+		int port = -1;
+		boolean rerunIDE = false;
+		if (context.getProperty(TESTER_CONTROLPORT) != null) {
+			port = Integer.parseInt(context.getProperty(TESTER_CONTROLPORT));
+			rerunIDE = true;
+		} else if (context.getProperty(TESTER_PORT) != null) {
+			port = Integer.parseInt(context.getProperty(TESTER_PORT));
+		}
+
+		if (port > 0) {
+			try {
+				trace("using control port %s, rerun IDE?: %s", port, rerunIDE);
+				jUnitEclipseListener = new JUnitEclipseListener(port, rerunIDE);
+				listenerList.add(jUnitEclipseListener);
+			} catch (Exception e) {
+				System.err.println(
+					"Cannot create link Eclipse JUnit control on port " + port + " (rerunIDE: " + rerunIDE + ')');
+				System.exit(254);
+			}
+		}
+
+		String testerDir = context.getProperty(TESTER_DIR);
+		if (testerDir == null)
+			testerDir = "testdir";
+
+		reportDir = new File(testerDir);
+
+		//
+		// Jenkins does not detect test failures unless reported
+		// by JUnit XML output. If we have an unresolved failure
+		// we timeout. The following will test if there are any
+		// unresolveds and report this as a JUnit failure. It can
+		// be disabled with -testunresolved=false
+		//
+		unresolved = context.getProperty(TESTER_UNRESOLVED);
+
+		trace("run unresolved %s", unresolved);
+
+		if (!reportDir.exists() && !reportDir.mkdirs()) {
+			System.err.printf("Could not create directory %s%n", reportDir);
+		} else {
+			trace("using %s, path: %s", reportDir, reportDir.toPath());
+			try {
+				listenerList
+					.add(new LegacyXmlReportGeneratingListener(reportDir.toPath(), new PrintWriter(System.err)));
+			} catch (Exception e) {
+				error("Error trying to create xml reporter: %s", e);
+			}
+		}
+
+		basicReport = LoggingListener.forBiConsumer((t, msg) -> {
+			if (t == null) {
+				trace(msg.get());
+			} else {
+				trace(msg.get(), t);
+			}
+		});
+		summary = new SummaryGeneratingListener();
+		listenerList.add(basicReport);
+		listenerList.add(summary);
+		listeners = listenerList.stream()
+			.toArray(TestExecutionListener[]::new);
+
+		if (testcases == null) {
+			trace("automatic testing of all bundles with " + aQute.bnd.osgi.Constants.TESTCASES + " header");
+			try {
+				automatic();
+			} catch (IOException e) {
+				// ignore
+			}
+		} else {
+			trace("receivednames of classes to test %s", testcases);
+			try {
+				baseSelectors = Stream.of(testcases.split("\\s*,\\s*"))
```

Comment:
Whenever you want to `.split(String)` a string, Use `Strings.splitAsStream` instead.  This code must compile the pattern, split the the string into an array and then stream the array. `Strings.splitAsStream` already has the precompiled comma-separator pattern and splits the string directly into a stream.

Reply:
Done, thanks. I think that was the last one - I did a search of the project for "split" and this was the only one that came up. Let me know if you stumble across any others.